config/v1/types_cluster_version.go

```diff
@@ -215,6 +222,58 @@ type UpdateHistory struct {
 // ClusterID is string RFC4122 uuid.
 type ClusterID string
 
+// ClusterVersionCapabilitiesDefault is a constant representing the
+// default policy for optional, core cluster components.
+type ClusterVersionCapabilitiesDefault string
+
+const (
+	// IncludeClusterVersionCapabilities indicates that optional,
+	// core cluster components should be included by default.
+	IncludeClusterVersionCapabilities = "Include"
+
+	// ExcludeClusterVersionCapabilities indicates that optional,
+	// core cluster components should be excluded by default.
+	ExcludeClusterVersionCapabilities = "Exclude"
+)
+
+// ClusterVersionCapabilitiesSpec configures the installation of
+// optional, core cluster components.
+// +k8s:deepcopy-gen=true
+type ClusterVersionCapabilitiesSpec struct {
+	// inclusionDefault declares the default policy for optional,
+	// core cluster components.
+
+	// +optional
+	InclusionDefault ClusterVersionCapabilitiesDefault `json:"inclusionDefault,omitempty"`
+
+	// include declares capabilities that should be included,
+	// regardless of inclusionDefault.
+	// +listType=atomic
+	// +optional
+	Include []string `json:"include,omitempty"`
```

Comment:
I bet I know the answer, but is there any way to verify items in this list?

Reply:
yes, and the expectation is that the CVO will do so, and complain if you set an invalid value.  There has been some recent discussion about a centrally-managed registry of valid caps, and holding that in the API repo as an enum would allow the API server to block typos and such automatically.

Reply:
As of 06f44f95, we have an enum with registered values in this PR, which should give us API-server validation.  Although I might need to read up on kubebuilder docs to get the Go enum values into the CRD...

Reply:
ok, I've added two `+kubebuilder:validation:Enum`, so now everything should have CRD-side guards enforcing the API-side registries.  Marking this thread resolved.